blocks/library/image/index.js

```diff
@@ -59,6 +59,8 @@ registerBlockType( 'core/image', {
 		},
 	},
 
+	capability: 'upload_files',
```

Comment:
Not sure we should apply the capability to the whole block in this case. It should be ok for a user to pick an existing image from the library without uploading. Seems the components themselves (media upload button, draggable upload area) are the ones that should check for the capability and change how they are displayed.

Reply:
Thanks a lot for your review @mtias 

On one hand I totally agree a contributor should be able to pick an existing media from the WordPress Media Library.

On the other hand, he can't even do that 😢  Being logged in as a contributor, when you click on the button to pick an image, a video or an audio file, you get an empty library as the ajax action that performs the attachments query is checking for the `upload_files` capability very early. See [/wp-admin/includes/ajax-actions.php](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/includes/ajax-actions.php#L2679).

So today these "media" blocks are pretty useless for contributors, that's why I suggested to remove them as it's an experience that is a bit frustrating 😬 

To only allow picking to contributors would require a WordPress core change that would be a bit more difficult than simply allowing uploading and picking just like the other roles, imho. I can think of UI/feature changes to remove the Uploader & the Upload tab of the WordPress Media Editor for instance.
Adding the `upload_files` capability to the contributor role would be much easier, but it would probably require some wide communication to WordPress users as I can imagine some are using this role for the users they wan't to restrict from uploading files...

Reply:
This makes sense. I think it'd be ok to remove them from the inserter, but how should we handle editing an existing post that has images? Or a post that has an image placeholder?

Reply:
> how should we handle editing an existing post that has images? Or a post that has an image placeholder?

@imath Is there anything which needs to be considered here?

Reply:
@aduth This is handled into the PR actually. Images or any other media in existing post will be displayed. Just like in the classic editor the contributor can remove the image from the post content (but not from the Media Library) or edit display attributes (alignment, etc..)